### DISCUSSION TITLE
initialState更改问题

### DISCUSSION THREAD

**Discussion by midsmr** [umijs/umi#12011] [Q&A]
`initialState`在`getInitialState`方法中被初始化后，只能在组件中使用`useModel`进行更新吗？

我想在`app.tsx`中`layout`配置项`onPageChange`更改`initialState`，请问大家可以吗？

**Comment by liujiayii** [comment]
按道理useModel只能在function组件里用，可以试试这个https://umijs.org/docs/max/valtio

**Comment by fz6m** [comment]
如同楼上所述，可以尝试使用 [valtio](https://umijs.org/docs/max/valtio) 等可以脱离 react hook 的全局数据流方案。
